src/datastore/StringCrypto.ts

```diff
@@ -48,19 +66,66 @@ export class StringCrypto {
     }
 
     public encrypt(plaintext: string): string {
-        const salt = crypto.randomBytes(16).toString('base64');
-        return crypto.publicEncrypt(
+        const salt = randomBytes(SALT_BYTE_LENGTH).toString(SALT_ENCODING);
+        return publicEncrypt(
             this.privateKey,
             Buffer.from(salt + ' ' + plaintext)
-        ).toString('base64');
+        ).toString(ENCRYPTED_ENCODING);
     }
 
     public decrypt(encryptedString: string): string {
-        const decryptedPass = crypto.privateDecrypt(
+        const decryptedPass = privateDecrypt(
             this.privateKey,
-            Buffer.from(encryptedString, 'base64')
+            Buffer.from(encryptedString, ENCRYPTED_ENCODING)
         ).toString();
         // Extract the password by removing the prefixed salt and seperating space
-        return decryptedPass.split(' ')[1];
+        return decryptedPass.slice(SALT_STRING_LENGTH + 1);
+    }
+
+    public async encryptLargeString(plaintext: string): Promise<string> {
+        const password = randomBytes(32).toString(ENCRYPTED_ENCODING);
+        const key = await scrypt(password, 'salt', 32) as Buffer;
```

Comment:
As long as you're generating a random password, there's probably not much of a point in passing to scrypt, as it can be used directly as an AES key.